src/hooks.ts

```diff
@@ -0,0 +1,314 @@
+import { FormatModule, LocalArgumentDefinition } from "relay-compiler";
+import addAnyTypeCast from "./addAnyTypeCast";
+import relayCompilerLanguageTypescript from "./index";
+import { loadCompilerOptions } from "./loadCompilerOptions";
+
+export default function relayHooksTypescriptCompiler() {
+  const compilerOptions = loadCompilerOptions();
+
+  const formatModule: FormatModule = (opts) => {
+    const {
+      // moduleName,
+      documentType,
+      docText,
+      concreteText,
+      typeText,
+      hash,
+      sourceHash,
+      definition,
+      definition: { name, metadata },
+    } = opts;
+
+    const allHooks: string[] = [];
+    const typeImports: string[] = [];
+    const reactImports: string[] = [];
+    const reactRelayImports: string[] = [];
+    const relayRuntimeImports: string[] = [];
+
+    if (documentType) {
+      relayRuntimeImports.push(documentType);
+    }
+
+    const meta = (metadata ?? {}) as CompilerMeta;
+
+    if (!meta.derivedFrom) {
```

Comment:
`derivedFrom` is defined from Relay's split operation transformer that runs as a part of it's overall query transform operation. It's either the name of the node the referenced node is derivedFrom or undefined.

So, if it's derived no hooks are generated. 

I get all that. I don't think I understand when it's classified as derived vs not. The transform looks like it runs on all operations. I guess only the base most nodes would be considered not derived?

Reply:
IIRC I believe that this is was for refetchable fragments: https://github.com/facebook/relay/blob/d59a32bcdbc11341f972def8e5967408a2131c18/packages/relay-compiler/language/javascript/RelayFlowGenerator.js#L989-L1006 essentially if it's "derived" then it's the operation name that will be called when refetching, but we don't actually want to create a `use{...}Query` definition for it because that's already encapsulated in the fragment.

From the example app:

```graphql
fragment TodoAppData on User
  @refetchable(queryName: "TodoAppRefetchQuery")
  @argumentDefinitions(
    last: { type: "Int" }
    first: { type: "Int" }
    after: { type: "String" }
    before: { type: "String" }
  ) {
    id
    totalCount
    isAppending
    ...TodoListFooterData
    ...TodoList
      @arguments(last: $last, first: $first, after: $after, before: $before)
}
```

We don't actually want to generate a `useTodoAppRefetchQuery` hook... that's just an internal  implementation detail of the `const [data, refetch] = useTodoAppData(data)` fragment hook.

And that's why nothing is output at the bottom of this file here:

https://github.com/tgriesser/relay-compiler-language-typescript/blob/c18a58f2cd2f82bb0c6f5b5bc6875779ff9d33e8/example-hooks-gen/ts/__relay_artifacts__/TodoAppRefetchQuery.graphql.ts#L410-L422

Reply:
Cool, thanks!